Formula/consul.rb

```diff
@@ -71,10 +83,29 @@ def plist
   end
 
   test do
+    http_port = free_port
     fork do
-      exec "#{bin}/consul", "agent", "-data-dir", "."
+      # most ports must be free, but are irrelevant to this test
+      system(
+        "#{bin}/consul",
+        "agent",
+        "-dev",
+        "-bind", "127.0.0.1",
+        "-dns-port", "-1",
+        "-grpc-port", "-1",
+        "-http-port", http_port,
+        "-serf-lan-port", free_port,
+        "-serf-wan-port", free_port,
+        "-server-port", free_port
```

Comment:
Are these required for the test?

Reply:
Yes, unfortunately they're non-optional, cannot seem to be disabled (like e.g. the RPC port) and the defaults clash with any other Consul process which happens to be using them, so we must choose an arbitrary free port.

Reply:
Okay, lgtm then.